local.go

```diff
@@ -54,59 +54,100 @@ func Local(f Function, initX []float64, settings *Settings, method Method) (*Res
 	}
 
 	startTime := time.Now()
-
 	funcs, funcInfo := getFunctionInfo(f)
+	status := NotTerminated
+
+	if funcInfo.IsStatuser {
+		_, err := funcs.status.Status()
+		if err != nil {
+			// Function status is bad, panic.
+			// TODO: Document this panic in the function comment.
+			panic(fmt.Sprintf("bad initial function status: %v", err))
```

Comment:
The motivation for this check was to make sure that the function has a good status before calling getStartingLocation(). The motivation for the panic was that if we return an error, the caller should be able to handle it, so if he is able to handle it after, he can also handle it before and should not call Local() with bad function status in the first place. I do not insist on this, I put in there as a possibility and to discuss it.

I will change it to "return nil, err".